common/hello.go

```diff
@@ -0,0 +1,190 @@
+				// Meaning that it won't be retried until the next chart-operator reconciliation loop 5 minutes later.
+				// To speed things up, we keep patching the hello-world `App` CR by adding a label. That way, we trigger reconciliation loops in chart-operator.
+				now := time.Now()
+				patchedApp := helloApp.DeepCopy()
+				labels := patchedApp.GetLabels()
+				labels["update"] = fmt.Sprintf("%d", now.Unix())
+				patchedApp.SetLabels(labels)
+				managementClusterKubeClient := state.GetFramework().MC()
+				err := managementClusterKubeClient.Patch(ctx, patchedApp, ctrl.MergeFrom(helloApp))
+				if err != nil {
+					return err
+				}
+				app, err := state.GetFramework().GetApp(ctx, helloApp.Name, helloApp.Namespace)
+				if err != nil {
+					return err
+				}
+				return checkAppStatus(app)
+			}).
+				WithTimeout(6 * time.Minute).
+				WithPolling(5 * time.Second).
+				Should(Succeed())
+		})
+
+		It("hello world app responds successfully", func() {
+			if !externalDnsSupported {
+				Skip("Bastion is not supported.")
+			}
+
+			Eventually(func() (*http.Response, error) {
+				helloWorldIngressUrl := fmt.Sprintf("https://%s", helloWorldIngressHost)
+				logger.Log("Trying to get a successful response from %s", helloWorldIngressUrl)
+				return http.Get(helloWorldIngressUrl)
+			}, "10m", "5s").Should(HaveHTTPStatus(http.StatusOK))
+		})
+
+		AfterEach(func() {
+			if !externalDnsSupported {
+				Skip("Bastion is not supported.")
+			}
+
+			ctx := context.Background()
+
+			managementClusterKubeClient := state.GetFramework().MC()
+			err := managementClusterKubeClient.Delete(ctx, nginxApp)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = managementClusterKubeClient.Delete(ctx, nginxConfigMap)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = managementClusterKubeClient.Delete(ctx, helloApp)
+			Expect(err).ShouldNot(HaveOccurred())
+			err = managementClusterKubeClient.Delete(ctx, helloConfigMap)
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+	})
+}
+
+// deployApp creates an `App` CR for the desired application.
+// It currently has some needed workarounds until we improve our `clustertest` framework.
+func deployApp(ctx context.Context, name, namespace string, organization *organization.Org, version, valuesFile string, values map[string]string) (*v1alpha1.App, *v1.ConfigMap) {
+	var err error
+	managementClusterKubeClient := state.GetFramework().MC()
+	appBuilder := application.New(fmt.Sprintf("%s-%s", state.GetCluster().Name, name), name).
+		WithCatalog("giantswarm").
+		WithOrganization(*organization).
+		// If we don't pass any version value when building the app, the latest version will be calculated and used. But the calculation will fail when the app name doesn't match the repository name.
+		WithVersion(version).
+		WithInCluster(false).
+		// We need to manually set this label that should be automatically added once we can set the `app.Config` property.
```

Comment:
Do we have an issue for `app.Config`?

Reply:
Not yet, I want to create tickets for all the workarounds documented here after this is merged.

Reply:
https://github.com/giantswarm/giantswarm/issues/28067